packages/store/src/blob/types.ts

```diff
@@ -0,0 +1,24 @@
+import type { Signal } from '../utils/signal';
+
+export type BlobId = string;
+export type BlobURL = string;
+
+export interface BlobProvider {
+  blobs: Set<BlobId>;
+  signals: {
+    blobAdded: Signal<BlobId>;
+  };
+  get(id: BlobId): Promise<BlobURL | null>;
+  set(blob: Blob): Promise<BlobId>;
+  delete(id: BlobId): Promise<void>;
+  clear(): Promise<void>;
+}
+
+export type IdbInstance<T = ArrayBufferLike> = {
+  get: (key: BlobId) => Promise<T | undefined>;
```

Comment:
* `IdbInstance` -> `IDBInstance` 
* Can we use `<T | null>` instead of `<T | undefined>`?

Reply:
idb-keyval define type with Promise<T | undefined>, additional runtime check is required if we wish to convert correctly